core/arch/x86/decode.c

```diff
@@ -1809,6 +1824,41 @@ decode_operand(decode_info_t *di, byte optype, opnd_size_t opsize, opnd_t *opnd)
         /* no need to set size as it's a GPR */
         return true;
     }
+    case TYPE_K_MODRM: {
+        /* part of AVX-512: modrm.rm selects opmask register or mem addr */
+        if (di->mod != 3) {
+            return decode_modrm(di, optype, opsize, NULL, opnd);
+        }
+        /* fall through*/
+    }
+    case TYPE_K_MODRM_R: {
+        /* part of AVX-512: modrm.rm selects opmask register */
+        *opnd = opnd_create_reg(decode_reg(DECODE_REG_RM, di, optype, opsize));
+        /* For VEX encoded scalar opmask instructions, the size is determined
+         * by the opcode.
```

Comment:
I don't understand this comment: what is it telling the reader?  We have the opcode-determined size here in the `opsize` variable.  Should we call opnd_set_size(opnd, opsize)?  Do we need this comment?  Ditto below.

Reply:
I thought it helps to understand that the opmask instruction defines the k register's size implicitly by its opcode. Removed the comment(s), done.